### PR TITLE
The basic outline for the document entities and associated factories

### PR DESCRIPTION
**Pull request by N1cholasMcKee**

### CODE REVIEW

**Comment by HumrajB** [comment]
Great work! The outlines look very complete and well documented. One thing I see is that you have repeated javadoc for the getNumberOfQuestions method in the TestDocument Class on line 68-71. Just remove this and it should be good to merge into main!

File: src/main/java/entities/TestDocument.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * Gets the allotted time for the test
+     * @return The time for the test
+     */
+    public String getEstimatedTime() {
+        return estimatedTime;
+    }
+
+    /**
```
Comment (HumrajB) [inline src/main/java/entities/TestDocument.java:68]: Repeated Javadoc for getNumberOfQuestions.

**Review by HumrajB — COMMENTED** [review]
Great work! The outlines look very complete and well documented. One thing I see is that you have repeated javadoc for the getNumberOfQuestions method in the TestDocument Class on line 68-71. Just remove this and it should be good to merge into main!

**Review by takedakento — COMMENTED** [review]
All looks good to me. Well organized and documented code, and a good foundation for handling the actual tests and solutions :)

**Review by HarveyRDonnelly — APPROVED** [review]
Looks great. Comment issue can be fixed in later PR.

**Pull request merged**